src/main/java/com/dokev/gold_dduck/event/repository/EventRepository.java

```diff
@@ -1,7 +1,17 @@
 package com.dokev.gold_dduck.event.repository;
 
 import com.dokev.gold_dduck.event.domain.Event;
+import com.dokev.gold_dduck.gift.domain.Gift;
+import java.util.List;
+import java.util.UUID;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface EventRepository extends JpaRepository<Event, Long> {
+
+    Event findEventByCode(@Param("code") UUID code);
+
+    @Query(value = "SELECT a FROM Gift AS a WHERE a.event = :eventId")
```

Comment:
```suggestion
    @Query(value = "SELECT g FROM Gift AS g WHERE g.event = :eventId")
```
a보단 Gift니까 g가 더 좋을 것 같습니다.